Game/RainMeadow.EntityHooks.cs

```diff
@@ -29,6 +30,17 @@ private void EntityHooks()
 
             On.OverseerAI.UpdateTempHoverPosition += OverseerAI_UpdateTempHoverPosition; // no teleporting
             On.OverseerAI.Update += OverseerAI_Update; // please look at what i tell you to
+
+            On.UpdatableAndDeletable.Destroy += UpdatableAndDeletable_Destroy;
+        }
+
+        private void UpdatableAndDeletable_Destroy(On.UpdatableAndDeletable.orig_Destroy orig, UpdatableAndDeletable self)
+        {
+            //Adding a delay on mark for garbage collection. Need to give network a chance to see an object has been destroyed
+            Task.Run(async () => {
+                await Task.Delay(250);
+                orig(self);
+            });
```

Comment:
I disagree, you're "waiting for state to sync" while instead this should be handled as an event.
Is the entity system not picking this up as an "entity removed" event?